auth/api/iam/api.go

```diff
@@ -382,25 +386,37 @@ func (r Wrapper) GetRequestJWT(ctx context.Context, request GetRequestJWTRequest
 	ro := new(jarRequest)
 	err := r.authzRequestObjectStore().Get(request.Id, ro)
 	if err != nil {
-		return nil, err
+		return nil, oauth.OAuth2Error{
+			Code:        oauth.InvalidRequest,
+			Description: "request object not found",
+		}
 	}
 	// compare raw strings, don't waste a db call to see if we own the request.Did.
 	if ro.Client.String() != request.Did {
-		return nil, errors.New("invalid request")
+		return nil, oauth.OAuth2Error{
+			Code:          oauth.InvalidRequest,
+			Description:   "client_id does not match request",
+			InternalError: errors.New("DID does not match client_id for requestID"),
+		}
 	}
 	if ro.RequestURIMethod != "get" {
 		// TODO: wallet does not support `request_uri_method=post`. Signing the current jarRequest would leave it without 'aud'.
-		//		 is this acceptable or should it fail?
+		//		 is this acceptable, should it fail, or does it default to using staticAuthorizationServerMetadata.
 		return nil, oauth.OAuth2Error{
 			Code:          oauth.InvalidRequest,
 			Description:   "used request_uri_method 'get' on a 'post' request_uri",
 			InternalError: errors.New("wrong 'request_uri_method' authorization server or wallet probably does not support 'request_uri_method'"),
 		}
 	}
+
+	// TODO: supported signature types should be checked
 	token, err := r.jar.Sign(ctx, ro.Claims)
 	if err != nil {
-		// TODO: oauth.OAuth2Error?
-		return nil, err
+		return nil, oauth.OAuth2Error{
+			Code:          oauth.ServerError,
+			Description:   "failed to sign authorization RequestObject",
```

Comment:
not sure if you want to convey this, since it's details about the internal error condition. Could just be "unable to create RequestObject"